x-pack/plugins/ml/public/application/components/jobs_awaiting_node_warning/new_job_awaiting_node_shared/new_job_awaiting_node_shared.tsx

```diff
@@ -0,0 +1,150 @@
+}
+
+function isJobAwaitingNodeAssignment(job: estypes.MlJobStats) {
+  return job.node === undefined && job.state === JOB_STATE.OPENING;
+}
+
+const MLJobsAwaitingNodeWarning: FC<Props> = ({ jobIds }) => {
+  const { http } = useKibana().services;
+  const ml = useMemo(() => mlApiServicesProvider(new HttpService(http!)), [http]);
+
+  const [unassignedJobCount, setUnassignedJobCount] = useState<number>(0);
+  const [cloudInfo, setCloudInfo] = useState<CloudInfo | null>(null);
+
+  const checkNodes = useCallback(async () => {
+    try {
+      if (jobIds.length === 0) {
+        setUnassignedJobCount(0);
+        return;
+      }
+
+      const { lazyNodeCount } = await ml.mlNodeCount();
+      if (lazyNodeCount === 0) {
+        setUnassignedJobCount(0);
+        return;
+      }
+
+      const { jobs } = await ml.getJobStats({ jobId: jobIds.join(',') });
+      const unassignedJobs = jobs.filter(isJobAwaitingNodeAssignment);
+      setUnassignedJobCount(unassignedJobs.length);
+    } catch (error) {
+      setUnassignedJobCount(0);
+      // eslint-disable-next-line no-console
+      console.error('Could not determine ML node information', error);
+    }
+  }, [jobIds]);
+
+  const checkCloudInfo = useCallback(async () => {
+    if (unassignedJobCount === 0) {
+      return;
+    }
+
+    try {
+      const resp = await ml.mlInfo();
+      const cloudId = resp.cloudId ?? null;
+      setCloudInfo({
+        isCloud: cloudId !== null,
+        cloudId,
+        deploymentId: cloudId === null ? null : extractDeploymentId(cloudId),
+      });
+    } catch (error) {
+      setCloudInfo(null);
+      // eslint-disable-next-line no-console
+      console.error('Could not determine cloud information', error);
+    }
+  }, [unassignedJobCount]);
+
+  useEffect(() => {
+    checkCloudInfo();
+  }, [unassignedJobCount]);
+
+  useEffect(() => {
+    checkNodes();
+  }, [jobIds]);
+
+  if (unassignedJobCount === 0) {
+    return null;
+  }
+
+  return (
+    <>
+      <EuiCallOut
+        title={
+          <FormattedMessage
+            id="xpack.ml.jobsAwaitingNodeWarningShared.title"
+            defaultMessage="Awaiting machine learning node"
+          />
+        }
+        color="primary"
+        iconType="iInCircle"
+      >
+        <div>
+          <FormattedMessage
+            id="xpack.ml.jobsAwaitingNodeWarningShared.noMLNodesAvailableDescription"
+            defaultMessage="There {jobCount, plural, one {is} other {are}} {jobCount, plural, one {# job} other {# jobs}} waiting for machine learning nodes to start."
+            values={{
+              jobCount: unassignedJobCount,
+            }}
+          />
+          <EuiSpacer size="s" />
+          {cloudInfo &&
+            (cloudInfo.isCloud ? (
+              <>
+                <FormattedMessage
+                  id="xpack.ml.jobsAwaitingNodeWarningShared.isCloud"
+                  defaultMessage="Elastic Cloud will autoscale to add more ML capacity. This may take 5-20 minutes. "
```

Comment:
Since it seems like there are potentially some deployments that have autoscale limitations (e.g. https://www.elastic.co/guide/en/cloud-enterprise/current/ece-autoscaling.html#ece-autoscaling-restrictions) and folks can feasibly disable autoscaling, so IMO "can" is preferable to "will:
```suggestion
                  defaultMessage="Elastic Cloud deployments can autoscale to add more ML capacity. This may take 5-20 minutes. "
```

Reply:
updated in https://github.com/elastic/kibana/pull/112264/commits/29e50eb96da70c6697c63356ed41018e3b9d239e